### PR TITLE
feat(groovy-common): Phase 8 - GroovyDoc Parser

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary
- Adds `GroovyDoc` data classes for structured documentation representation
- Implements `GroovyDocParser` with full tag parsing support
- Parses @param, @return, @throws, @see, @since, @deprecated, @author tags
- Handles multiline descriptions and comment syntax stripping

## Test plan
- [x] All 21 unit tests passing
- [x] Spotless formatting applied
- [x] Module builds successfully

---

> [!NOTE]
> Introduces structured GroovyDoc parsing to convert comment blocks into typed metadata.
> 
> - Adds `GroovyDoc` and tag models (`ParamTag`, `ReturnTag`, `ThrowsTag`, `SeeTag`) for structured docs
> - Implements `GroovyDocParser` to strip comment syntax, split description vs. tags, and parse `@param`, `@return(s)`, `@throws`/`@exception`, `@see`, `@since`, `@deprecated`, `@author`
> - Supports multiline descriptions and multiline tag content aggregation
> - Adds unit tests covering empty/single-line comments, multiline descriptions, multiple tags, alternative tag names, missing descriptions, and mixed scenarios
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 929c3fbc48e2de068ba08980c3d634529579cf2c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Release Notes

* **New Features**
  * Added GroovyDoc parsing capability to extract and structure documentation comments from Groovy code, with support for descriptions, parameters, return values, thrown exceptions, related references, and metadata tags (since, deprecated, author).

* **Tests**
  * Comprehensive test suite covering various documentation parsing scenarios and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-common/src/main/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParser.kt [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            .dropLastWhile { it.isEmpty() || it == "*/" || it == "*" }
+            .map { line ->
+                when {
+                    line.startsWith("/**") -> line.removePrefix("/**").trim()
+                    line.startsWith("*/") -> ""
+                    line.startsWith("*") -> line.removePrefix("*").trimStart()
+                    else -> line
+                }
+            }
+            .filter { it.isNotEmpty() || it.isBlank() } // Keep blank lines for multiline content
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-common/src/main/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParser.kt]: Fixed in 929c3fbc

[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-common/src/main/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParser.kt]: Fixed in 929c3fbc

File: groovy-common/src/main/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParser.kt [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            emptyList()
+        } else {
+            parseTagLines(lines.subList(tagStartIndex, lines.size))
+        }
+
+        return buildGroovyDoc(description, tags)
+    }
+
+    /**
+     * Removes comment syntax (slash-star-star and star-slash, and leading *) from each line.
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-common/src/main/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParser.kt]: Fixed in 929c3fbc

File: groovy-common/src/test/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParserTest.kt [hunk 3]
```diff
@@ -0,0 +1,391 @@
+package com.github.albertocavalcante.groovycommon.groovycommon.doc
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-common/src/test/kotlin/com/github/albertocavalcante/groovycommon/doc/GroovyDocParserTest.kt]: Fixed in 929c3fbc
